File: examples/cxx/suspend/LazyGuy.cxx
#include "LazyGuy.hpp"
#include <Host.hpp>

#include <Msg.hpp>

// Not needed : DreamMaster directly constructs the object
// MSG_IMPLEMENT_DYNAMIC(LazyGuy, Process)

int LazyGuy::main(int argc, char** argv)
{
	info("Hello !");
	
	info("Nobody's watching me ? Let's go to sleep.");
	
	Process::currentProcess().suspend();
	
	info("Uuuh ? Did somebody call me ?");
	
	info("Mmmh, goodbye now."); 

	delete this;

	return 0;
}

File: examples/suspend/LazyGuy.java
package suspend;

import org.simgrid.msg.Host;
import org.simgrid.msg.Msg;
import org.simgrid.msg.Process;
import org.simgrid.msg.MsgException;
public class LazyGuy extends Process {
	public LazyGuy(Host host, String name, String[]args) {
		super(host,name,args);
	} 
	public void main(String[] args) throws MsgException {
		Msg.info("Nobody's watching me ? Let's go to sleep.");
		suspend();
		Msg.info("Uuuh ? Did somebody call me ?");
		Msg.info("Mmmh, goodbye now.");
	}
}